tasks/src/main/scala/dagr/tasks/fgbio/ExtractUmisFromBam.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package dagr.tasks.fgbio
+
+import dagr.commons.CommonsDef._
+
+import scala.collection.mutable.ListBuffer
+
+class ExtractUmisFromBam(val in: PathToBam,
+                         val out: PathToBam,
+                         val rs1: String,
+                         val rs2: String,
+                         val tags1: Seq[String],
+                         val tags2: Seq[String],
+                         val annotateNames: Boolean = true
+                        ) extends FgBioTask {
+
+  protected def addFgBioArgs(buffer: ListBuffer[Any]): Unit = {
+    buffer.append("-i", in)
+    buffer.append("-o", out)
+    buffer.append("-r", rs1)
+    buffer.append("-R", rs2)
+    tags1.foreach(buffer.append("-b", _))
```

Comment:
Probably a matter of taste, but I think I would write:
```scala
if (tags1.nonEmpty) {
    buffer.append("-b")
    buffer.append(tags1:_*)
}
```
So that it would produce `-b ZA ZB` instead of `-b ZA -b ZB`.